### ISSUE TITLE
Do some package have something build in your own system? Like a building error showing your path.

### ISSUE THREAD

**Issue by Ericonaldo** [sisl/ngsim_env#20]
The setting are strictly follows your guide and the system is ubuntu16.04.
The errors are as follows:

(rllab3) xxx@XXX:~/ngsim_env/scripts/imitation$ python validate.py --n_proc 1 --exp_dir ../../data/experiments/NGSIM-gail/ --params_filename itr_1000.npz --random_seed 42
50 vehicles with H = 200
2019-07-03 10:57:27.242208: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
pid: 0 traj: 58 / 2062
signal (11): Segmentation fault
in expression starting at no file:0
GetResult at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Modules/_ctypes/callproc.c:911 [inlined]
_ctypes_callproc at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Modules/_ctypes/callproc.c:1186
PyCFuncPtr_call at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Modules/_ctypes/_ctypes.c:3856
PyObject_Call at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Objects/abstract.c:2165
do_call at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:4936 [inlined]
call_function at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:4732 [inlined]
PyEval_EvalFrameEx at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:3236
fast_function at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:4803 [inlined]
call_function at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:4730 [inlined]
PyEval_EvalFrameEx at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:3236
fast_function at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:4803 [inlined]
call_function at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:4730 [inlined]
PyEval_EvalFrameEx at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:3236
fast_function at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:4803 [inlined]
call_function at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:4730 [inlined]
PyEval_EvalFrameEx at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:3236
fast_function at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:4803 [inlined]
call_function at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:4730 [inlined]
PyEval_EvalFrameEx at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:3236
_PyEval_EvalCodeWithName at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:4018
PyEval_EvalCodeEx at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:4039
function_call at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Objects/funcobject.c:627
PyObject_Call at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Objects/abstract.c:2165
method_call at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Objects/classobject.c:330
PyObject_Call at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Objects/abstract.c:2165
PyObject_CallFunctionObjArgs at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Objects/abstract.c:2445
call_attribute at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Objects/typeobject.c:6089 [inlined]
slot_tp_getattr_hook at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Objects/typeobject.c:6131
builtin_getattr at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/bltinmodule.c:998
PyCFunction_Call at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Objects/methodobject.c:109
call_function at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:4705 [inlined]
PyEval_EvalFrameEx at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:3236
_PyEval_EvalCodeWithName at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:4018
PyEval_EvalCodeEx at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Python/ceval.c:4039
function_call at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Objects/funcobject.c:627
PyObject_Call at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Objects/abstract.c:2165
method_call at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Objects/classobject.c:330
PyObject_Call at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Objects/abstract.c:2165
PyObject_CallFunctionObjArgs at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Objects/abstract.c:2445
call_attribute at /home/ilan/minonda/conda-bld/work/Python-3.5.2/Objects/typeobject.c:6089 [inlined]

**Comment by ZhilaiShen** [comment]
Hi Ericonaldo, I am actually having the exact same problem as yours. Have you been able to solve this problem?

**Comment by nebneBgnahZ** [comment]
Same issue here.

**Comment by nebneBgnahZ** [comment]
> Hi Ericonaldo, I am actually having the exact same problem as yours. Have you been able to solve this problem?



> Hi Ericonaldo, I am actually having the exact same problem as yours. Have you been able to solve this problem?

same issue here.

**Comment by nebneBgnahZ** [comment]
> Hi Ericonaldo, I am actually having the exact same problem as yours. Have you been able to solve this problem?

same issue here.

**Comment by Ericonaldo** [comment]
No... But maybe it is not the path problem. I think it is the error that the authors had mentioned [here](https://github.com/sisl/ngsim_env/blob/master/scripts/imitation/README.md), which says "this is somehow related to julia processes remaining unfinished and the python script moving on. Looking in validate.py, there is a sleep() call"

**Comment by ZhilaiShen** [comment]
And I think he also mentions the [Seg fault](ngsim_env), which is solved by deleting the PyCall.jl, but for julia1.1, there is no longer such folder.
Another issue is that the issue only occurs when I use my own dataset, while the original NGSIM dataset is still working. Are you using the original dataset or your own dataset?

**Comment by Ericonaldo** [comment]
Did you work well with the NGSIM dataset that the authors packaged up? I encountered this error when I use the packaged NGSIM dataset.

**Comment by nebneBgnahZ** [comment]
> Did you work well with the NGSIM dataset that the authors packaged up? I encountered this error when I use the packaged NGSIM dataset.

I also encountered this error using the NGSIM dataset. I am wondering if anything was wrong when I created the observations. In addition, I also encountered error when I run the bash script "run_experiment.sh". Apparently all the models for multi-agent failed. But I do have some results for single-agent scenario.

**Comment by ZhilaiShen** [comment]
When I try to run it with packaged NGSIM data, the only problem I encountered is that n_env from 10 to 20, the Tensorflow alerts because the parameters of layers are not cleared, and not allowed. 
BTW, do you guys have this problem occurring "randomly"?

**Comment by ZhilaiShen** [comment]
Hi all
I am not sure if this works for all, but I think the problem can be solved by removing the cache file under folder "~/julia/.julia/compiled/v1.1/PyCall/****.jl". It can be the same problem [here](https://github.com/sisl/ngsim_env/blob/master/docs/install.md), under section trouble shooting, reason 1. The cache file path for julia 1.1 is different from path in julia 0.6, so I delete the new cache file, and it seems to be working.

**Comment by Ericonaldo** [comment]
> Hi all
> I am not sure if this works for all, but I think the problem can be solved by removing the cache file under folder "~/julia/.julia/compiled/v1.1/PyCall/****.jl". It can be the same problem [here](https://github.com/sisl/ngsim_env/blob/master/docs/install.md), under section trouble shooting, reason 1. The cache file path for julia 1.1 is different from path in julia 0.6, so I delete the new cache file, and it seems to be working.

Hi, ZhilaiShen
You can work the whole thing now?

**Comment by ZhilaiShen** [comment]
sorry Minghuan, not really

________________________________

**Comment by raks097** [comment]
@Ericonaldo  @ZhilaiShen  @nebneBgnahZ 
I am currently facing the same issue,
Were any of you able to get this working?

**Comment by Ericonaldo** [comment]
No. Actually I already gave it up ...

**Comment by DarrenRuan** [comment]
Have you tried to use `single_process_collect_trajectories` rather than `parallel_collect_trajectories`? @Ericonaldo @raks097 @ZhilaiShen @nebneBgnahZ Then we might be able to skip the part (line 190 in validate.py):

```python
# let the julia processes finish up
    time.sleep(10)
```

Although it might be much slower.

Edit: Well. Although I choose to use `single_process_collect_trajectories`, I still met that weird Segmentation fault.
